src/CLI/CLI.cs

```diff
@@ -89,7 +95,15 @@ static void ShowHelp(Mono.Options.OptionSet options)
             Console.WriteLine("   contain only the bindings for that header file.");
         }
 
-        static void HandleOutputArg(String arg)
+        static void AddIncludeDirs(String dir, ref List<String> messages)
```

Comment:
List<> is a class and there's no need to pass classes with "ref" because they are passed by reference anyway.
Please remove "ref" in all places where you pass lists, not just here.